src/repository.c

```diff
@@ -598,94 +665,90 @@ int git_repository_open_bare(
 }
 
 static int _git_repository_open_ext_from_env(
-	git_repository **out,
+	struct repo_env *env,
 	const char *start_path,
-	const git_repository_open_options *opts)
+	git_repository_open_options *opts)
```

Comment:
So instead of opening the repository, we now just try to find all necessary paths and return that instead? Shouldn't the function be renamed in that case?

Reply:
Fixed, I've renamed two offending functions to clarify what's being setup from what (the new `struct repo_env`, and this one).